RecoHGCal/TICL/plugins/TrackstersMergeProducer.cc

```diff
@@ -528,9 +539,34 @@ void TrackstersMergeProducer::produce(edm::Event &evt, const edm::EventSetup &es
     }
   }
 
+  // Compute timing
+  computeTime(*resultCandidates);
```

Comment:
You are looping once again over all `resultCandidates`.

I think it could be more in the spirit of this code (and more adherent to the name of the method itself) if `computeTime(TICLCandidate &)` return a `std::pair<float, float>` which contains the time and its error for the given candidate. Then you can call it at the end od the previous loop (before `resultCandidates->push_back(tmpCandidate)`) and `if (timeErr>0)`  you can `setTime` and `setTimeError` there

Reply:
There are 10 calls to `resultCandidates->push_back(tmpCandidate)` in that code, that's why I added the loop right before the storing of `resultCandidates` into the event.

Reply:
You are right, ok for staying as such

> There are 10 calls to `resultCandidates->push_back(tmpCandidate)` in that code, that's why I added the loop right before the storing of `resultCandidates` into the event.

You are right @lecriste , ok for staying as such